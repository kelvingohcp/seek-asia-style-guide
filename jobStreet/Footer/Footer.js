import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';
import FooterLinks from './components/FooterLinks/FooterLinks';
import { getLocalization } from '../localization';
import _get from 'lodash/get';

const Footer = ({ language, country }) => {
  const year = new Date().getFullYear();
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <footer className={styles.container}>
        <FooterLinks footerMessages={footerMessages} />
        <p className={styles.copyright}>
          {_get(footerMessages, 'copyright').replace('{year}', year)}
        </p>
      </footer>
    );
  }

  return null;
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
