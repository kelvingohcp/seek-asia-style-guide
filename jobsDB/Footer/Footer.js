import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';

import FooterLinks from './components/FooterLinks/FooterLinks';
import FooterHK from './FooterHK/FooterHK';
import localization from './localization';

const Footer = ({ language, country, ...restProps }) => {
  const year = new Date().getFullYear();
  const messages = localization[`${language}-${country}`];

  if (country === 'hk') {
    return (<FooterHK language={language} country={country} {...restProps} />);
  }
  return (
    <footer className={styles.container}>
      <div className={styles.wrapper}>
        <div className={styles.pullRight}>
          <FooterLinks messages={messages} />
        </div>
        <div className={styles.pullLeft}>
          <p className={styles.copyright}>
            {messages['footer.copyright'].replace('{year}', year)}
            <span className={styles.fullversion}>
              {messages['footer.rightReserved']}
            </span>
          </p>
        </div>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
