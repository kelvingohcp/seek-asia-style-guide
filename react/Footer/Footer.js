import React from 'react';
import PropTypes from 'prop-types';
import GlobalFooter from '../private/GlobalFooter/GlobalFooter';
import { getLocalization } from '../../jobsDB/localization';
import { getLocalList } from '../../jobsDB/helper';
import _get from 'lodash/get';

const Footer = ({
  language,
  country,
  isExpandedVersion,
  showHeaderActionTrayOffset,
  showCountryLanguage,
  domainUrl,
  ...restProps }) => {
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <GlobalFooter
        country={country}
        isExpandedVersion={isExpandedVersion}
        showHeaderActionTrayOffset={showHeaderActionTrayOffset}
        showCountryLanguage={showCountryLanguage}
        footerMessages={footerMessages}
        locales={getLocalList({ messages, domainUrl, country, language })}
        {...restProps}
      />
    );
  }
  return null;
};

Footer.defaultProps = {
  country: 'hk',
  language: 'en',
  domainUrl: 'jobsdb.com',
  hasCompanyProfile: true,
  showHeaderActionTrayOffset: true,
  isExpandedVersion: false,
  showCountryLanguage: true
};

Footer.propTypes = {
  country: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  domainUrl: PropTypes.string.isRequired,
  hasCompanyProfile: PropTypes.bool,
  showHeaderActionTrayOffset: PropTypes.bool,
  isExpandedVersion: PropTypes.bool,
  showCountryLanguage: PropTypes.bool
};

export default Footer;
