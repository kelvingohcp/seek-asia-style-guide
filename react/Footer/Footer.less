@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "~seek-asia-style-guide/react/Header/components/ActionTray/ActionTray.less";
@wrapper-up-padding-top: @grid-base * 8;
@wrapper-up-padding-bottom: @grid-base * 3;

.container {
  cursor: default;
  margin: 0 auto;
  background-color: @saGrey6;
  width: 100%;
  font-weight: @sa-semi-bold;
}

.headerActionTrayOffset {
  @media @mobile {
    margin-bottom: @actionTrayHeight;
  }
}

.link {
  text-decoration: none;
}

.upperWrapper {
  padding-top: @wrapper-up-padding-top;
  padding-bottom: @wrapper-up-padding-bottom;

  -webkit-transition: all 0.3s ease-in-out;
  -moz-transition: all 0.3s ease-in-out;
  -ms-transition: all 0.3s ease-in-out;
  -o-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;

  &.hidden {
    max-height: 0;
    padding-top: 0;
    padding-bottom: 0;
    overflow: hidden;
    opacity: 0;
  }
}

.lowerWrapper {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
}

.lowerWrapperLinks {
  display: flex;
  flex-direction: column;
  line-height: @grid-base * 4;
  padding-left: @grid-base * 2;
  width: 100%;
  @media @tablet {
    flex-direction: row;
    line-height: normal;
    padding-bottom: @grid-base;
    width: 100%;
  }
  @media @desktop {
    padding-bottom: @grid-base;
    width: auto;
  }
}

.fullWidth {
  width: 100%;
}

.collapsed {
  padding-left: @grid-base * 4;
  @media @desktop {
    padding-left: @grid-base * 6;
  }
}

.chevronSvg {
  height: @grid-base * 3;
  width: @grid-base * 3;
}

.chevronIcon {
  cursor: pointer;
  padding-right: @grid-base * 4;
  padding-left: @grid-base * 8;
  display: flex;
  align-items: center;
  color: @saBlue2;
  @media @desktop {
    padding-right: @grid-base * 6;
    padding-left: @grid-base * 12;
  }
}

.endOfLine {
  display: flex;
  width: 100%;
  @media @tablet {
    width: 100%;
  }
  @media @desktop {
    width: auto;
  }
}

.endOfLineWithoutCountry {
  display: flex;
  width: 100%;
}

.localeIcon {
  margin: @grid-base @grid-base 0 0;
}

.currentLocale {
  display: none;
  @media @desktop {
    display: flex;
  }
}
