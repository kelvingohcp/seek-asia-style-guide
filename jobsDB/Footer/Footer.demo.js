import Footer from './Footer';
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as sketch from './Footer.sketch';

export const makeDummyLinkRendererForPath = path => {
  const DummyLinkRenderer = ({ href, children, ...props }) => (
    <Link style={{ textDecoration: 'none' }} to={`${path}#actualHref=${href}`} {...props}>
      {children}
    </Link>
  );

  DummyLinkRenderer.propTypes = {
    children: PropTypes.array,
    href: PropTypes.string
  };

  return DummyLinkRenderer;
};

const ROUTE = '/jobsdb-footer';

export default {
  route: ROUTE,
  tenantPath: 'jobsDB',
  title: 'JobsDB Footer',
  component: Footer,
  sketch,
  initialProps: {
    language: 'en',
    country: 'hk',
    linkRenderer: makeDummyLinkRendererForPath(ROUTE)
  },
  options: [
    {
      label: 'Language',
      type: 'radio',
      states: [
        {
          label: 'English',
          transformProps: props => ({
            ...props,
            language: 'en'
          })
        },
        {
          label: 'Bahasa',
          transformProps: props => ({
            ...props,
            language: 'id'
          })
        },
        {
          label: 'Thai',
          transformProps: props => ({
            ...props,
            language: 'th'
          })
        }
      ]
    },
    {
      label: 'Country',
      type: 'radio',
      states: [
        {
          label: 'Hong Kong',
          transformProps: props => ({
            ...props,
            country: 'hk'
          })
        },
        {
          label: 'Singapore',
          transformProps: props => ({
            ...props,
            country: 'sg'
          })
        },
        {
          label: 'Indonesia',
          transformProps: props => ({
            ...props,
            country: 'id'
          })
        },
        {
          label: 'Thailand',
          transformProps: props => ({
            ...props,
            country: 'th'
          })
        }
      ]
    }
  ]
};
