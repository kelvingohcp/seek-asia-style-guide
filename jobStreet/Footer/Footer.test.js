import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('Footer component', () => {
  const _Date = Date;

  beforeEach(() => {
    const DATE = new Date('2017');
    global.Date = jest.fn(() => DATE);
    global.Date.getFullYear = _Date.getFullYear;
  });

  afterEach(() => {
    global.Date = _Date;
  });

  describe('Localization', () => {
    it('should render en-my', () => {
      const wrapper = shallow(<Footer language="en" country="my" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-id', () => {
      const wrapper = shallow(<Footer language="en" country="id" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-ph', () => {
      const wrapper = shallow(<Footer language="en" country="ph" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-sg', () => {
      const wrapper = shallow(<Footer language="en" country="sg" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-vn', () => {
      const wrapper = shallow(<Footer language="en" country="vn" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render id-id', () => {
      const wrapper = shallow(<Footer language="id" country="id" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render vi-vn', () => {
      const wrapper = shallow(<Footer language="vi" country="vn" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
