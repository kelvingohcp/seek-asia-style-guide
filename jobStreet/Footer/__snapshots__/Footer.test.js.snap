// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Localization should render en-id 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.co.id/en/about-us/",
        "footer.aboutLinkText": "About Us",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.co.id/home/feedback.php?site=id&sub=feedback",
        "footer.feedbackLinkText": "Send Feedback",
        "footer.helpLink": "https://jobstreet.co.id/en/user/",
        "footer.helpLinkText": "Help",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet International",
        "footer.privacyLink": "https://jobstreet.co.id/en/about-us/privacy-policy/",
        "footer.privacyLinkText": "Privacy Policy",
        "footer.profilesLink": "https://jobstreet.co.id/en/companies/778589-jobstreet",
        "footer.profilesLinkText": "Work With Us",
        "footer.safeLink": "https://jobstreet.co.id/en/about-us/safe-job-search-guide/",
        "footer.safeLinkText": "Safe Job Search Guide",
        "footer.siteMapLink": "https://jobstreet.co.id/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.co.id/en/about-us/terms-of-use/",
        "footer.termsLinkText": "Term of Use",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.co.id/en/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.co.id/en/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.co.id/en/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.co.id/en/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.co.id/en/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.co.id/en/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.co.id",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.co.id/en/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render en-my 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.com.my/en/about-us/",
        "footer.aboutLinkText": "About Us",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.com.my/home/feedback.php?site=my&sub=feedback",
        "footer.feedbackLinkText": "Send Feedback",
        "footer.helpLink": "https://jobstreet.com.my/en/user/",
        "footer.helpLinkText": "Help",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet International",
        "footer.privacyLink": "https://jobstreet.com.my/en/about-us/privacy-policy/",
        "footer.privacyLinkText": "Privacy Policy",
        "footer.profilesLink": "https://jobstreet.com.my/career/JobStreet.htm",
        "footer.profilesLinkText": "Work With Us",
        "footer.safeLink": "https://jobstreet.com.my/en/about-us/safe-job-search-guide/",
        "footer.safeLinkText": "Safe Job Search Guide",
        "footer.siteMapLink": "https://jobstreet.com.my/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.com.my/en/about-us/terms-of-use/",
        "footer.termsLinkText": "Term of Use",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.com.my/en/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.com.my/en/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.com.my/en/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.com.my/en/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.com.my/en/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.com.my/en/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.com.my",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.com.my/en/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render en-ph 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.com.ph/en/about-us/",
        "footer.aboutLinkText": "About Us",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.com.ph/home/feedback.php?site=ph&sub=feedback",
        "footer.feedbackLinkText": "Send Feedback",
        "footer.helpLink": "https://jobstreet.com.ph/en/user/",
        "footer.helpLinkText": "Help",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet International",
        "footer.privacyLink": "https://jobstreet.com.ph/en/about-us/privacy-policy/",
        "footer.privacyLinkText": "Privacy Policy",
        "footer.profilesLink": "https://www.jobstreet.com.ph/en/companies/507168-jobstreet-philippines-inc-ph",
        "footer.profilesLinkText": "Work With Us",
        "footer.safeLink": "https://jobstreet.com.ph/en/about-us/safe-job-search-guide/",
        "footer.safeLinkText": "Safe Job Search Guide",
        "footer.siteMapLink": "https://jobstreet.com.ph/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.com.ph/en/about-us/terms-of-use/",
        "footer.termsLinkText": "Term of Use",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.com.ph/en/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.com.ph/en/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.com.ph/en/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.com.ph/en/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.com.ph/en/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.com.ph/en/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.com.ph",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.com.ph/en/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render en-sg 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.com.sg/en/about-us/",
        "footer.aboutLinkText": "About Us",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.com.sg/home/feedback.php?site=sg&sub=feedback",
        "footer.feedbackLinkText": "Send Feedback",
        "footer.helpLink": "https://jobstreet.com.sg/en/user/",
        "footer.helpLinkText": "Help",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet International",
        "footer.privacyLink": "https://jobstreet.com.sg/en/about-us/privacy-policy/",
        "footer.privacyLinkText": "Privacy Policy",
        "footer.profilesLink": "https://jobstreet.com.sg/career/JobStreet.htm",
        "footer.profilesLinkText": "Work With Us",
        "footer.safeLink": "https://jobstreet.com.sg/en/about-us/safe-job-search-guide/",
        "footer.safeLinkText": "Safe Job Search Guide",
        "footer.siteMapLink": "https://jobstreet.com.sg/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.com.sg/en/about-us/terms-of-use/",
        "footer.termsLinkText": "Term of Use",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.com.sg/en/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.com.sg/en/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.com.sg/en/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.com.sg/en/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.com.sg/en/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.com.sg/en/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.com.sg",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.com.sg/en/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render en-vn 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.vn/en/about-us/",
        "footer.aboutLinkText": "About Us",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.vn/home/feedback.php?site=vn&sub=feedback",
        "footer.feedbackLinkText": "Send Feedback",
        "footer.helpLink": "https://jobstreet.vn/en/user/",
        "footer.helpLinkText": "Help",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet International",
        "footer.privacyLink": "https://jobstreet.vn/en/about-us/privacy-policy/",
        "footer.privacyLinkText": "Privacy Policy",
        "footer.profilesLink": "https://jobstreet.vn/en/companies/762582-jobstreet-company-limited-vn",
        "footer.profilesLinkText": "Work With Us",
        "footer.safeLink": "https://jobstreet.vn/en/about-us/safe-job-search-guide/",
        "footer.safeLinkText": "Safe Job Search Guide",
        "footer.siteMapLink": "https://jobstreet.vn/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.vn/en/about-us/terms-of-use/",
        "footer.termsLinkText": "Term of Use",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render id-id 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.co.id/id/about-us/",
        "footer.aboutLinkText": "Tentang Kami",
        "footer.copyright": "Hak Cipta © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.co.id/home/feedback.php?site=id&sub=feedback",
        "footer.feedbackLinkText": "Kirim Saran",
        "footer.helpLink": "https://jobstreet.co.id/id/user/",
        "footer.helpLinkText": "Bantuan",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet Internasional",
        "footer.privacyLink": "https://jobstreet.co.id/id/about-us/kebijakan-privasi/",
        "footer.privacyLinkText": "Kebijakan Kerahasiaan",
        "footer.profilesLink": "https://jobstreet.co.id/en/companies/778589-jobstreet",
        "footer.profilesLinkText": "Berkarier bersama Kami",
        "footer.safeLink": "https://jobstreet.co.id/id/about-us/panduan-dan-tips-mencari-kerja-online-secara-aman/",
        "footer.safeLinkText": "Panduan Mencari Kerja Secara Aman",
        "footer.siteMapLink": "https://jobstreet.co.id/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.co.id/id/about-us/ketentuan-penggunaan/",
        "footer.termsLinkText": "Ketentuan Penggunaan",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.co.id/id/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.co.id/id/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.co.id/id/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.co.id/id/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.co.id/id/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.co.id/id/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.co.id",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.co.id/id/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Hak Cipta © 2017 JobStreet.com
  </p>
</footer>
`;

exports[`Footer component Localization should render vi-vn 1`] = `
<footer
  className="container"
>
  <FooterLinks
    messages={
      Object {
        "footer.aboutLink": "https://jobstreet.vn/vi/about-us/",
        "footer.aboutLinkText": "Về JobStreet",
        "footer.copyright": "Copyright © {year} JobStreet.com",
        "footer.feedbackLink": "https://myjobstreet.jobstreet.vn/home/feedback.php?site=vn&sub=feedback",
        "footer.feedbackLinkText": "Gửi Phản Hồi",
        "footer.helpLink": "https://jobstreet.vn/vi/user/",
        "footer.helpLinkText": "Trợ Giúp",
        "footer.link": "https://www.jobstreet.com",
        "footer.linkText": "JobStreet Quốc Tế",
        "footer.privacyLink": "https://jobstreet.vn/vi/about-us/chinh-sach-bao-mat/",
        "footer.privacyLinkText": "Chính Sách Bảo Mật",
        "footer.profilesLink": "https://jobstreet.vn/vi/companies/762582-jobstreet-company-limited-vn",
        "footer.profilesLinkText": "Cơ Hội Nghề Nghiệp",
        "footer.safeLink": "https://jobstreet.vn/vi/about-us/huong-dan-va-bi-quyet-tim-viec-truc-tuyen-an-toan/",
        "footer.safeLinkText": "Mẹo Tìm Việc An Toàn",
        "footer.siteMapLink": "https://jobstreet.vn/sitemap.htm",
        "footer.siteMapLinkText": "Site Map",
        "footer.termsLink": "https://jobstreet.vn/vi/about-us/dieu-khoan-su-dung/",
        "footer.termsLinkText": "Điều Khoản Sử Dụng",
        "header.careerInsightsTitle": "Career Insights",
        "header.careerInsightsUrl": "https://jobstreet.vn/vi/career-insights",
        "header.classifiedJobsTitle": "Classified Jobs",
        "header.classifiedJobsUrl": "https://jobstreet.vn/vi/job-search/classified-ads.php",
        "header.companyProfilesTitle": "Company Profiles",
        "header.companyProfilesUrl": "https://jobstreet.vn/vi/companies/",
        "header.educationTitle": "Education",
        "header.educationUrl": "https://jobstreet.vn/vi/career-insights",
        "header.employerLinkPrefix": "Looking to hire? Check out our ",
        "header.employerSiteTitle": "Employer Site",
        "header.employerSiteUrl": "https://jobstreet.vn/vi/cms/employer",
        "header.freshGradJobsTitle": "Fresh Grad Jobs",
        "header.freshGradJobsUrl": "https://jobstreet.vn/vi/job-search/campus/job-vacancy.php",
        "header.homeTitle": "Home",
        "header.homeUrl": "https://jobstreet.vn",
        "header.homeUrlText": "JOBSTREET",
        "header.loginTitle": "Log in",
        "header.loginUrl": "https://myjobstreet.jobstreet.com.my/login",
        "header.myAccountTitle": "My Account",
        "header.myAccountUrl": "https://myjobstreet.jobstreet.com.my/resume/preview-resume.php?",
        "header.myJobStreetTitle": "MyJobStreet",
        "header.myJobStreetUrl": "https://myjobstreet.jobstreet.com.my/home/login.php?site=my",
        "header.overseasJobsTitle": "Overseas Jobs",
        "header.overseasJobsUrl": "https://jobstreet.vn/vi/job-search/browse/find-overseas-jobs",
        "header.seperatorOr": "or",
        "header.signupTitle": "Sign up",
        "header.signupUrl": "https://myjobstreet.jobstreet.com.my/signup",
        "menu.backToMenu": "Back to Menu",
        "menu.countryAndLanguage": "Country & Language",
        "menu.jobSeekerHeader": "JOB SEEKER",
        "menu.more": "More",
        "menu.moreHeader": "MORE",
        "menu.settingsHeader": "SETTINGS",
      }
    }
  />
  <p
    className="copyright"
  >
    Copyright © 2017 JobStreet.com
  </p>
</footer>
`;
